Source/CocoaMQTTDeliver.swift

```diff
@@ -210,11 +213,11 @@ extension CocoaMQTTDeliver {
         
         let nowTimestamp = Date(timeIntervalSinceNow: 0).timeIntervalSince1970
         for (idx, frame) in inflight.enumerated() {
-            if (nowTimestamp - frame.timestamp) >= (retryTimeInterval/1000.0) {
+            if (nowTimestamp - frame.timestamp) >= (retryTimeInterval/1000.0) * Double(frame.retryCount) {
                 
                 var duplicatedFrame = frame
                 duplicatedFrame.frame.dup = true
-                duplicatedFrame.timestamp = nowTimestamp
```

Comment:
I don't believe we should remove the this part:

` duplicatedFrame.timestamp = nowTimestamp`